File: client/python/mesh/kinds/service.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from typing import Dict

from mesh.macro import index, serializable


@serializable
class Service:
    """
    Service definition.
            service = Service()
            service.urn = ""
            service.namespace = ""
            service.name = ""
            service.version = version
            service.proto = proto
            service.codec = codec
            service.asyncable = asyncable
            service.timeout = 3000
            service.retries = 3
            service.node = ""
            service.inst = ""
            service.zone = ""
            service.cluster = ""
            service.cell = ""
            service.group = ""
            service.address = ""
            service.kind = ""
            service.lang = ""
            service.attrs = {}
    """

    @index(0)
    def urn(self) -> str:
        """
        Uniform domain name.
        """
        pass

    @index(5)
    def namespace(self) -> str:
        """
        Service namespace.
        """
        pass

    @index(10)
    def name(self) -> str:
        """
        Service name.
        """
        pass

    @index(15)
    def version(self) -> str:
        """
        Service version.
        """
        pass

    @index(20)
    def proto(self) -> str:
        """
        Service net protocol.
        """
        pass

    @index(25)
    def codec(self) -> str:
        """
        Service payload codec protocol.
        """
        pass

    @index(30)
    def flags(self) -> int:
        """
        Service serve asyncable.
        """
        pass

    @index(35)
    def timeout(self) -> int:
        """
        Service serve timeout in mill seconds.
        """
        pass

    @index(40)
    def retries(self) -> int:
        """
        Service serve retry times.
        """
        pass

    @index(45)
    def node(self) -> str:
        """
        Service customized zone.
        """
        pass

    @index(50)
    def inst(self) -> str:
        """
        Service customized institution.
        """
        pass

    @index(55)
    def zone(self) -> str:
        """
        Service customized zone.
        """
        pass

    @index(60)
    def cluster(self) -> str:
        """
        Service customized cluster.
        """
        pass

    @index(65)
    def cell(self) -> str:
        """
        Service customized cell.
        """
        pass

    @index(70)
    def group(self) -> str:
        """
        Service customized group.
        """
        pass

    @index(75)
    def group(self) -> str:
        """
        Service customized sets.
        """
        pass

    @index(80)
    def address(self) -> str:
        """
        Service customized address.
        """
        pass

    @index(85)
    def kind(self) -> str:
        """
        Service kind.
        """
        pass

    @index(90)
    def lang(self) -> str:
        """
        Service lang.
        """
        pass

    @index(95)
    def attrs(self) -> Dict[str, str]:
        """
        Service customized attrs.
        """
        pass

File: client/java/src/main/java/com/be/mesh/client/struct/Service.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.struct;

import com.be.mesh.client.annotate.Binding;
import com.be.mesh.client.annotate.Bindings;
import com.be.mesh.client.annotate.Index;
import com.be.mesh.client.annotate.MPS;
import com.be.mesh.client.tool.LDCable;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serializable;
import java.util.Map;

/**
 * @author coyzeng@gmail.com
 */
@Data
@EqualsAndHashCode
public class Service implements Serializable, LDCable {

    private static final long serialVersionUID = -3867455899170650883L;
    /**
     * Service urn.
     */
    @Index(0)
    private String urn;
    /**
     * Service topic.
     */
    @Index(5)
    private String namespace;
    /**
     * Service topic.
     */
    @Index(10)
    private String name;
    /**
     * Service version.
     */
    @Index(15)
    private String version;
    /**
     * Net protocol.
     */
    @Index(20)
    private String proto;
    /**
     * Serialize protocol.
     */
    @Index(25)
    private String codec;
    /**
     * Service invoke asyncable.
     */
    @Index(30)
    private long flags;
    /**
     * Invoke timeout.
     */
    @Index(35)
    private long timeout;
    /**
     * Response retries.
     */
    @Index(40)
    private int retries;
    /**
     * Service node identity.
     */
    @Index(45)
    private String node;
    /**
     * Service inst identity.
     */
    @Index(50)
    private String inst;
    /**
     * Service zone.
     */
    @Index(55)
    private String zone;
    /**
     * Service cluster.
     */
    @Index(60)
    private String cluster;
    /**
     * Service cell.
     */
    @Index(65)
    private String cell;
    /**
     * Service group.
     */
    @Index(70)
    private String group;
    /**
     * Service sets.
     */
    @Index(75)
    private String sets;
    /**
     * Service address.
     */
    @Index(80)
    private String address;
    /**
     * Target uri. Service kind. see {@link MPS} {@link Binding} {@link Bindings}
     */
    @Index(85)
    private String kind;
    /**
     * Service lang.
     */
    @Index(90)
    private String lang;
    /**
     * Service arguments.
     */
    @Index(95)
    private Map<String, String> attrs;

}
